packages/medusa/src/models/discount.ts

```diff
@@ -79,10 +79,10 @@ export class Discount extends SoftDeletableEntity {
 
   @BeforeInsert()
   private upperCaseCode(): void {
+    this.code = this.code.toUpperCase()
```

Comment:
comment: @srindom In our discount factory, we provide an explicitly set an id leading to this function never being called

Reply:
👍